Assignment3/assignment3.txt

```diff
@@ -0,0 +1,12 @@
+Aldin Aksay
+
+1. Determine the complexity
+	a: 24 bytes, O(n)
+	b: 32 bytes, O(n^2 + n)
+	c: 24 bytes, O(1/2n * 2^n)
```

Comment:
It's definitely not exponential - if it were exponential, it would print for an incredibly long time. The first for loop gives a time complexity of n since you iterate through n elements. The second for loop would normally also be n, but since it multiplies the j by two each time, that means you're eliminating half the number of elements each time - this causes a time complexity of log (n). And since they're nested, you multiply the two with each other so that the time complexity is O(nlog(n)).